File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int maxWater = 0;
        int left = 0; 
        int right = height.length - 1;

        while(left < right) {
            int water = (right - left) * Math.min(height[left], height[right]);
            maxWater = Math.max(water, maxWater);

            if(height[left] < height[right]) {
                left++;
            } else {
                right--;
            }
        }    

        return maxWater;
    }
}

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left, right = 0, len(height) - 1
        max_water = float('-inf')

        while left < right:
            curr_water = min(height[left], height[right]) * (right - left)
            max_water = max(max_water, curr_water)

            if height[left] < height[right]:
                left += 1
            else:
                right -= 1
        
        return max_water